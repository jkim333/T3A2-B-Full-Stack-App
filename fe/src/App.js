import React from "react";
import "./App.css";
import Login from "./pages/Login";
import Signup from "./pages/Signup";
import { Routes, Route } from "react-router-dom";
import Mainhome from "./pages/Home";
import Search from "./pages/Exercise";
import Log from "./pages/Log";
import Startpage from "./pages/Start";

function App() {
  return (
    <>
      <Routes>
        <Route path="/" element={<Mainhome />} />
        <Route path="/signup" element={<Signup />} />
        <Route path="/start" element={<Startpage />} />
        <Route path="/login" element={<Login />} />
        <Route path="/log-entry" element={<Log />} />
        <Route path="/search-exercise" element={<Search />} />
      </Routes>
    </>
  );
}

export default App;
