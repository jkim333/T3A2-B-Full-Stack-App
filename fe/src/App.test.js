import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import App from './App';
import Navbar from './components/navbar'




//brandname
test('renders the Xero fitness brand name', () => {
    render(<Navbar/>)
    const element = screen.getByTestId('brandname')
    expect(element).toHaveTextContent('Xero fitness');
})


function fetchData(){
    const result = await fetch("http://localhost:3002/")
    const data = await result.json()
    return data
}

test('fetches exercise list from backend server',()=>{
     
     
})


