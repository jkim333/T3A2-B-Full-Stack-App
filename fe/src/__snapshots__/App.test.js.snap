// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="w-screen h-screen overflow-hidden box-border  bg-gradient-to-b from-blue-900 to-sky-800"
  >
    <div
      className="flex justify-between bg-sky-800"
    >
      <h1
        className="font-sans text-blue-200 basis-1/2 text-2xl sm:text-3xl md:text-4xl lg:text-6xl xl:text-7xl pl-3 py-1 xl:pl-12"
      >
        Xero fitness
      </h1>
      <button
        className="pr-3 cursor-pointer"
        onClick={[Function]}
      >
        <svg
          color="rgb(191 219 254)"
          fill="currentColor"
          height={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "rgb(191 219 254)",
            }
          }
          viewBox="0 0 512 512"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </button>
    </div>
    <p
      className="text-blue-200 text-2xl border-b border-blue-300  mt-5 text-center"
    >
      Exercise Log
    </p>
  </div>
</div>
`;
